### ISSUE TITLE
RandomMosaic: list-input box padding counts are not permuted with the tiles (lost boxes and phantom boxes)

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4715]
## Summary
With a **list** box input, `RandomMosaic` counts the padding of each output row using the wrong source images. `apply_transform_boxes` permutes `_box._data` by `params["permutation"]` (`kornia/augmentation/_2d/mix/mosaic.py`, `_box._data[perm[:, 0]] = _box._data[perm[:, _idx]]`), but it leaves `_box._N`, the per-sample padding count, unpermuted. When the result is exported back to a list, each output row therefore:
- **drops real boxes**, or
- **keeps padding rows as phantom boxes**, depending on which images were drawn.

## Reproduction (main @ 088586663, torch 2.14.0, CPU)
```python
import torch, kornia.augmentation as K
torch.manual_seed(0)
image = torch.rand(4, 1, 6, 8)
boxes = [torch.tensor([[1., 1., 3., 3.]]),
         torch.tensor([[1., 1., 3., 3.], [2., 1., 6., 5.], [0.5, 0.5, 2., 2.]]),
         torch.tensor([[1., 1., 4., 4.], [2., 2., 5., 5.]]),
         torch.tensor([[0., 0., 2., 2.]])]
aug = K.RandomMosaic(p=1.0, start_ratio_range=(0.0, 0.0), data_keys=["input", "bbox_xyxy"])
pipe = K.AugmentationSequential(aug, data_keys=["input", "bbox_xyxy"])
_, out = pipe(image, boxes)
print(aug._params["permutation"].tolist())   # [[0, 2, 2, 1], [1, 1, 3, 3], [2, 0, 0, 0], [3, 2, 1, 3]]
print([b.shape[0] for b in out])             # [4, 12, 8, 4]
```

| row | source images | real boxes (before clipping) | returned |
|---|---|---|---|
| 0 | 0, 2, 2, 1 | 1 + 2 + 2 + 3 = 8 | 4: real boxes lost |
| 1 | 1, 1, 3, 3 | 3 + 3 + 1 + 1 = 8 | 12: phantom rows such as `[7, 0, 7, 0]` |
| 2 | 2, 0, 0, 0 | 2 + 1 + 1 + 1 = 5 | 8: phantom rows |
| 3 | 3, 2, 1, 3 | 1 + 2 + 3 + 1 = 7 | 4: real boxes lost |

The counts are before clipping and area filtering, so a correct result may have fewer boxes, but it can never have more. Rows 1 and 2 return more than exist.

## Expected
Each tile's padding count should come from the source image that tile was drawn from: permute `_N` along with `_data`, or rebuild `_N` for each output row from the permutation. Tensor box inputs are not affected, because they carry no `_N`.

Found while re-reviewing #4679 (RandomMosaic boxes, #4652). The one-token mutation `merged if bool(selected)` → `0 if bool(selected)` in #4679's unselected-row padding survives because of this bug.

Posted on behalf of @ducha-aiki by Claude (Opus 5).
